### ISSUE TITLE
Local destination file locked up when downloading a non existing file from remote source.

### ISSUE THREAD

**Issue by volkstrader** [hgupta9/FluentFTP#77]
Local destination file was locked up because the output stream was not disposed properly.
FtpClient.cs line 2125
```
			// download the file straight to a file stream
			bool ok = DownloadFileInternal(remotePath, outStream);

			// close the file stream
			try {
				outStream.Dispose();
			} catch (Exception) { }
			return ok;
```
The non existing remote file caused DownloadFileInternal to throw an exception, which was not caught in DownloadFileToFile method, hence, the outStream skipped the dispose call, which locked up the local destination file.

Suggestion:
```
            using (outStream)
            {
                // download the file straight to a file stream
                return DownloadFileInternal(remotePath, outStream);
            }
```
outStream would never be NULL since it was assigned at line 2113, so it is safe to wrap outStream with using clause.

I also found the same pattern at the following lines:
line 2014
```
			// write the file onto the server
			bool ok = UploadFileInternal(fileStream, remotePath, createRemoteDir);

			// close the file stream
			try {
				fileStream.Dispose();
			} catch (Exception) { }
			return ok;
		}
```
could be refactored to 
```
            using (fileStream)
            {
                // write the file onto the server
                return UploadFileInternal(fileStream, remotePath, createRemoteDir);
            }
```
line 2049
```
			// write the file onto the server
			MemoryStream ms = new MemoryStream(fileData);
			ms.Position = 0;
			bool ok = UploadFileInternal(ms, remotePath, createRemoteDir);
			ms.Dispose();
			return ok;
```
could be refactored to 
```
            // write the file onto the server
            using (MemoryStream ms = new MemoryStream(fileData))
            {
                ms.Position = 0;
                return UploadFileInternal(ms, remotePath, createRemoteDir);
            }
```

**Comment by hgupta9** [comment]
Thank you for this suggestion. I'll implement it shortly.

**Comment by hgupta9** [comment]
Fixed in 16.4.1.

**Comment by volkstrader** [comment]
Thanks for the update.
